src/core/instance/instance.cpp

```diff
@@ -329,6 +329,19 @@ Instance *Instance::InitMultiple(uint8_t aIdx)
     return instance;
 }
 
+Instance &Instance::Get(uint8_t aIdx)
+{
+    void *instance = gMultiInstanceRaw + aIdx * INSTANCE_SIZE_ALIGNED;
+    return *static_cast<Instance *>(instance);
+}
+
+uint8_t Instance::GetIdx(Instance *aInstance)
+{
+    uint8_t idx = (reinterpret_cast<uint8_t *>(aInstance) - reinterpret_cast<uint8_t *>(gMultiInstanceRaw)) /
+                  INSTANCE_SIZE_ALIGNED;
+    return idx;
+}
```

Comment:
Use `static_cast` to make conversion to `uint8_t` explicit?
```suggestion
uint8_t Instance::GetIdx(Instance *aInstance)
{
    return static_cast<uint8_t>(
        (reinterpret_cast<uint8_t *>(aInstance) - reinterpret_cast<uint8_t *>(gMultiInstanceRaw)) /
        INSTANCE_SIZE_ALIGNED);
}
```